### PR TITLE
Use string-representation of Text Domain throughout codebase.

### PR DESCRIPTION
**Pull request by raamdev**
See websharks/zencache#576

### CODE REVIEW

**Comment by raamdev** [comment]
@jaswsinc Review please.

Two questions in particular: 

I assume these lines don't need to change, correct? 
- https://github.com/websharks/zencache/blob/000000-dev/zencache/zencache.inc.php#L2489-L2499

What does this do and is it still needed?
- https://github.com/websharks/zencache/blob/000000-dev/zencache/includes/share.php#L2357-L2378

File: zencache/includes/share.php [hunk 1]
```diff
@@ -235,7 +235,7 @@ public function __construct()
 				if(strpos(__NAMESPACE__, '\\') !== FALSE) // Sanity check.
 					throw new \exception('Not a root namespace: `'.__NAMESPACE__.'`.');
 
-				$this->slug = $this->text_domain = str_replace('_', '-', __NAMESPACE__);
+				$this->slug = 'zencache' = str_replace('_', '-', __NAMESPACE__);
```
Comment (raamdev) [inline zencache/includes/share.php]: @jaswsinc This line can probably be changed to `$this->slug = 'zencache';`, correct?

**Comment by jaswrks** [comment]
> I assume these lines don't need to change, correct?

You can remove that dynamic `$text_domain` and use `zencache` instead as a hard-coded string.

> What does this do and is it still needed?

This is a namespaced `__()` function. Since all of ZenCache is namespaced, every call that you see which uses `__()` actually uses that function instead. That function is a proxy for the root `\__()` function in WordPress. So if `\__()` exists, we use that. If not, we just do nothing and let the string pass through. This is needed in the early caching phase, because `\__()` does exist yet.

So this can stay as-is.

Example of the filtering that occurs with that in place:
`__('something, 'zencache')` fires `zencache\__()` fires `\__()` (if it exists)

**Comment by jaswrks** [comment]
I reviewed the larger set of changes. Looks great! :+1:

**Comment by raamdev** [comment]
> This is a namespaced `__()` function. [...] This is needed in the early caching phase, because `\__()` does exist yet.

Ah, thanks for explaining that. :-) 

> You can remove that dynamic $text_domain and use zencache instead as a hard-coded string.

Will do.

> I reviewed the larger set of changes. Looks great!

Awesome! Thanks for the review! :-)

**Pull request closed without merging**